diff --git a/ttl/NIF-Investigation.ttl b/ttl/NIF-Investigation.ttl
index 8759b02..6b967d5 100644
--- a/ttl/NIF-Investigation.ttl
+++ b/ttl/NIF-Investigation.ttl
@@ -23,8 +23,8 @@

 <http://ontology.neuinfo.org/NIF/ttl/NIF-Investigation.ttl> a owl:Ontology ;
     owl:imports <http://ontology.neuinfo.org/NIF/ttl/BIRNLex-OBI-proxy.ttl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/cogat_v0.3.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/CogPO.owl>,
+        <http://www.cognitiveatlas.org/ontology/cogat.owl>,
+        <http://www.cogpo.org/ontologies/CogPOver2010.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/NIF-Government-Granting-Agency.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/NIF-Resource.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/NIF-Scientific-Discipline.ttl>,
diff --git a/ttl/bridge/pr-bridge.ttl b/ttl/bridge/pr-bridge.ttl
index 983a3b3..41deaef 100644
--- a/ttl/bridge/pr-bridge.ttl
+++ b/ttl/bridge/pr-bridge.ttl
@@ -6,7 +6,7 @@
 @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

 <http://ontology.neuinfo.org/NIF/ttl/bridge/pr-bridge.ttl> a owl:Ontology ;
-    owl:imports <http://ontology.neuinfo.org/NIF/ttl/external/pr.owl>,
+    owl:imports <http://purl.obolibrary.org/obo/pr.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/pr-dead.ttl> .

 ### Serialized using the nifstd custom serializer v1.0.7
