diff --git a/ttl/NIF-Investigation.ttl b/ttl/NIF-Investigation.ttl
index 8b7d1cb..b01807d 100644
--- a/ttl/NIF-Investigation.ttl
+++ b/ttl/NIF-Investigation.ttl
@@ -22,8 +22,8 @@
 
 <http://ontology.neuinfo.org/NIF/ttl/NIF-Investigation.ttl> a owl:Ontology ;
     owl:imports <http://ontology.neuinfo.org/NIF/ttl/BIRNLex-OBI-proxy.ttl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/cogat_v0.3.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/CogPO.owl>,
+        <http://www.cognitiveatlas.org/ontology/cogat.owl>,
+        <http://www.cogpo.org/ontologies/CogPOver2010.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/NIF-Government-Granting-Agency.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/NIF-Resource.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/NIF-Scientific-Discipline.ttl>,
diff --git a/ttl/bridge/doid-bridge.ttl b/ttl/bridge/doid-bridge.ttl
index d8aa7ab..561c79c 100644
--- a/ttl/bridge/doid-bridge.ttl
+++ b/ttl/bridge/doid-bridge.ttl
@@ -7,7 +7,7 @@
 @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
 <http://ontology.neuinfo.org/NIF/ttl/bridge/doid-bridge.ttl> a owl:Ontology ;
-    owl:imports <http://ontology.neuinfo.org/NIF/ttl/external/doid.owl>,
+    owl:imports <http://purl.obolibrary.org/obo/doid.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/nif_backend.ttl> .
 
 ### Annotations
diff --git a/ttl/nif.ttl b/ttl/nif.ttl
index f8792cc..c2b7ad4 100644
--- a/ttl/nif.ttl
+++ b/ttl/nif.ttl
@@ -11,10 +11,10 @@
     owl:imports <http://ontology.neuinfo.org/NIF/ttl/bridge/chebi-bridge.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/bridge/doid-bridge.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/bridge/uberon-bridge.ttl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/ero.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/go.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/pr.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/uberon.owl>,
+        <http://purl.obolibrary.org/obo/ero.owl>,
+        <http://purl.obolibrary.org/obo/go.owl>,
+        <http://purl.obolibrary.org/obo/pr.owl>,
+        <http://purl.obolibrary.org/obo/uberon.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/ksdesc-defs.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/ncbigeneslim.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/NIF-Neuron-HBP-cell-import.ttl>,
diff --git a/ttl/nif_backend.ttl b/ttl/nif_backend.ttl
index 7a3c072..af73a62 100644
--- a/ttl/nif_backend.ttl
+++ b/ttl/nif_backend.ttl
@@ -16,9 +16,9 @@
 <http://ontology.neuinfo.org/NIF/ttl/nif_backend.ttl> a owl:Ontology ;
     owl:imports <http://ontology.neuinfo.org/NIF/ttl/BIRNLex-OBO-UBO.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/BIRNLex_annotation_properties.ttl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/pato.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/ro.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/ro_bfo1-1_bridge.owl>,
+        <http://purl.obolibrary.org/obo/pato.owl>,
+        <http://purl.obolibrary.org/obo/ro.owl>,
+        <http://purl.org/obo/owl/ro_bfo1-1_bridge.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/quality_bfo_bridge.ttl>,
         <http://www.geneontology.org/formats/oboInOwl>,
         <http://www.ifomis.org/bfo/1.1> ;
