diff --git a/ttl/nif.ttl b/ttl/nif.ttl
index f8792cc..c2b7ad4 100644
--- a/ttl/nif.ttl
+++ b/ttl/nif.ttl
@@ -11,10 +11,10 @@
     owl:imports <http://ontology.neuinfo.org/NIF/ttl/bridge/chebi-bridge.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/bridge/doid-bridge.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/bridge/uberon-bridge.ttl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/ero.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/go.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/pr.owl>,
-        <http://ontology.neuinfo.org/NIF/ttl/external/uberon.owl>,
+        <http://purl.obolibrary.org/obo/ero.owl>,
+        <http://purl.obolibrary.org/obo/go.owl>,
+        <http://purl.obolibrary.org/obo/pr.owl>,
+        <http://purl.obolibrary.org/obo/uberon.owl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/ksdesc-defs.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/ncbigeneslim.ttl>,
         <http://ontology.neuinfo.org/NIF/ttl/generated/NIF-Neuron-HBP-cell-import.ttl>,
